### ISSUE TITLE
How to add elements into sequence?

### ISSUE THREAD

**Issue by dotzhou** [pabigot/pyxb#63]
Dear Peter:

@pabigot 

I tried to use to BIND(element1, element2, ..., element) for a sequence complex type.

However it generate Nondeterminism exceeded validating errors.

How to create element first and append to a sequence? 

Could you please give me any tips or any references?

Best Wishes,

**Comment by dotzhou** [comment]
Dear Peter:

How to add elements to sequence complex type in order?

Best wishes

**Comment by dotzhou** [comment]
Dear Peter @pabigot

First, I tried to create a unordered sequenceType as below:

sequence.element_7 = "";
sequence.element_11 = ""
.. ... ..
sequence.element_n = "";

An unordered sequence is ready.

I wish to dump its contents to a ordered sequenceType as below:

orderedSequence = sequence.orderedContent()

How could I implement?

for element in sequence.orderedContent():
        orderedSequence.append(element)

Great thanks for your assistance.

**Comment by pabigot** [comment]
I don't understand your question.  It would be much easier if you would point to a gist that includes the schema you are using, examples of the code you have tried,  and what you want the result to be.

You can assign the member elements in whatever order you want; if the schema requires them to be in a specific order  they will be ordered when the instance is converted to DOM.

[This method](http://pyxb.sourceforge.net/api/pyxb.binding.basis.complexTypeDefinition-class.html#orderedContent) might also be relevant.

**Comment by dotzhou** [comment]
Dear Peter @pabigot:

Great thanks for your kind responses.

I tried to get around of "add elements to a sequence".

Two ways: 

1. BIND with correct types and order.
2. Append elements in correct sequence.

Your Pyxb even is suitable for large project for complex schema.

Could I ask another question?

How to solve encoding issues for characters as French or special ISO 8859-1 symbols as
?

I tried as below:
    
    text = "-80 to +20 °C : ±(0.226- 0.0028)mm"
    notes.append(text)

Failed

I set up as below:

    pyxb._InputEncoding = 'ISO-8859-1'
    pyxb._OutputEncoding = 'ISO-8859-1'

and Tried to output as below:

    contents = element.toDOM(element_name="geo:element").toprettyxml(indent='    ', 
                                                encoding='ISO-8859-1')


I could try convert "ISO-8859-1" to "UTF-8" be encoding as:


     text = "80 to +20 &#176;C : &#177;(0.226- 0.0028)mm"
     notes.append(text)

However, I believe you definitely have better and more elegant solutions?

Best wishes,

Ted
